lib/fs.js

```diff
@@ -543,6 +543,9 @@ function read(fd, buffer, offset, length, position, callback) {
 
   validateOffsetLengthRead(offset, length, buffer.byteLength);
 
+  if (typeof position === 'bigint')
```

Comment:
This seems to be a check that can be performed early instead of being this late? Before `validateBuffer` would be a good place I think.

Reply:
Wouldn't it be better to have the checks in order?

Reply:
My concern was mostly about the `if (length === 0)` above which means if the length is 0 you'd be able to get away with these checks..but thinking again it was already the case for a few other cases above, so it's probably not a bit deal.